Tests are failing

## Reproduction steps

Execute `make test` 

```
node-dom-extractor/src/dom-extractor.js:9
__cov_HZncuREk_uNlp2ZluOgJaw.s['1']++;const utils=require('./dom-utils');__cov_HZncuREk_uNlp2ZluOgJaw.s['2']++;const cheerio=require('cheerio');__cov_HZncuREk_uNlp2ZluOgJaw.s['3']++;const request=require('request');__cov_HZncuREk_uNlp2ZluOgJaw.s['4']++;const juice=require('juice');__cov_HZncuREk_uNlp2ZluOgJaw.s['5']++;const NodeCache=require('node-cache');__cov_HZncuREk_uNlp2ZluOgJaw.s['6']++;const cssom=require('cssom');__cov_HZncuREk_uNlp2ZluOgJaw.s['7']++;const url=require('url');__cov_HZncuREk_uNlp2ZluOgJaw.s['8']++;const extractor=module.exports;__cov_HZncuREk_uNlp2ZluOgJaw.s['9']++;const nodeCache=new NodeCache();function cleanRelativePathes($,absoluteUrl){__cov_HZncuREk_uNlp2ZluOgJaw.f['1']++;__cov_HZncuREk_uNlp2ZluOgJaw.s['11']++;$('link').each(()=>{__cov_HZncuREk_uNlp2ZluOgJaw.s['12']++;$(this).attr('href',utils.relToAbs(absoluteUrl,$(this).attr('href')));});__cov_HZncuREk_uNlp2ZluOgJaw.s['13']++;$('img').each(()=>{__cov_HZncu

TypeError: Cannot read property 'http://www.camif.fr/espace-service/livraison-gratuite.html#.contentpconseil#csstrue#innerTextfalse' of undefined
    at Object.extractor.fetch (/node-dom-extractor/src/dom-extractor.js:9:6507)
    at Object.<anonymous> (/node-dom-extractor/test/sample.js:46:14)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Object.Module._extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/istanbul/lib/hook.js:109:37)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at /usr/local/lib/node_modules/mocha/lib/mocha.js:222:27
    at Array.forEach (native)
    at Mocha.loadFiles (/usr/local/lib/node_modules/mocha/lib/mocha.js:219:14)
    at Mocha.run (/usr/local/lib/node_modules/mocha/lib/mocha.js:487:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/mocha/bin/_mocha:459:18)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Object.Module._extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/istanbul/lib/hook.js:109:37)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Function.Module.runMain (module.js:605:10)
    at runFn (/usr/local/lib/node_modules/istanbul/lib/command/common/run-with-cover.js:122:16)
    at /usr/local/lib/node_modules/istanbul/lib/command/common/run-with-cover.js:251:17
    at /usr/local/lib/node_modules/istanbul/lib/util/file-matcher.js:68:16
    at /usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:52:16
    at /usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:361:13
    at /usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:52:16
    at done (/usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:246:17)
    at /usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:44:16
    at /usr/local/lib/node_modules/istanbul/node_modules/async/lib/async.js:358:17
    at LOOP (fs.js:1615:14)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
make: *** [cover] Error 1
```

--- a/src/dom-extractor.js
+++ b/src/dom-extractor.js
@@ -120,14 +120,10 @@ extractor.fetch = (data, options, callback) => {
   if (isValidUrl) {
     cacheKey = `${data}#${options.selector}#css${options.inlineCss}#innerText${options.innerText}`;
     const cachedValue = nodeCache.get(cacheKey);
-
-    if (cachedValue[cacheKey] !== undefined) {
+    if (cachedValue && cachedValue[cacheKey] !== undefined) {
       callback(cachedValue[cacheKey]);
       return;
     }
-  }
-
-  if (isValidUrl) {
     if (data.indexOf('http') !== 0) {
       data = `http://${data}`;
     }

--- a/src/dom-extractor.js
+++ b/src/dom-extractor.js
@@ -71,7 +71,7 @@ function cssCallback($, options, callback) {
     nodeCache.set(cacheKey, juice.juiceDocument($, {
       extraCss: options.extraCss || ''
     })('body').html());
-    callback(nodeCache.get(cacheKey)[cacheKey]);
+    callback(nodeCache.get(cacheKey));
   } catch (e) {
     callback();
   }

--- a/test/sample.js
+++ b/test/sample.js
@@ -1,10 +1,11 @@
 const domextractor = require('./../src/dom-extractor');
+const assert = require('assert');
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
   + "<a href='ok'>ok mec</a></div><div class='c'>ok</div></body></html>",
   '.b',
   (data) => {
-    console.log('A:', data);
+    assert.notEqual(data, undefined);
   });
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
@@ -12,7 +13,7 @@ domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><
     selector: '.b',
     innerText: true
   }, (b) => {
-    console.log('B:', b);
+    assert.notEqual(b, undefined);
   });
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
@@ -21,7 +22,7 @@ domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><
     innerText: true,
     inlineCss: false
   }, (c) => {
-    console.log('C:', c);
+    assert.notEqual(c, undefined);
   });
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
@@ -30,7 +31,7 @@ domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><
     innerText: false,
     inlineCss: false
   }, (d) => {
-    console.log('D:', d);
+    assert.notEqual(d, undefined);
   });
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
@@ -40,7 +41,7 @@ domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><
     inlineCss: false,
     removeLinks: true
   }, (e) => {
-    console.log('E:', e);
+    assert.notEqual(e, undefined);
   });
 
 domextractor.fetch('http://www.camif.fr/espace-service/livraison-gratuite.html', {
@@ -48,5 +49,5 @@ domextractor.fetch('http://www.camif.fr/espace-service/livraison-gratuite.html',
   innerText: false,
   inlineCss: true
 }, (f) => {
-  console.log('F:', f);
+  assert.notEqual(f, undefined);
 });

--- a/test/sample.js
+++ b/test/sample.js
@@ -1,4 +1,4 @@
-const domextractor = require('..');
+const domextractor = require('./../src/dom-extractor');
 
 domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><body><div class='b'>"
   + "<a href='ok'>ok mec</a></div><div class='c'>ok</div></body></html>",
@@ -44,7 +44,7 @@ domextractor.fetch("<html><head><style>.b{background-color:#eee}</style></head><
   });
 
 domextractor.fetch('http://www.camif.fr/espace-service/livraison-gratuite.html', {
-  selector: '.contentpconseil',
+  selector: 'div.contentpconseil',
   innerText: false,
   inlineCss: true
 }, (f) => {

